test/e2e/internal/utils/prepare.go

```diff
@@ -115,5 +115,12 @@ func copyFile(srcFile, dstFile string) error {
 	defer from.Close()
 
 	_, err = io.Copy(to, from)
+	if err != nil {
+		return err
+	}
+
+	// Also preserve file permissions.
+	srcStat, _ := from.Stat()
+	err = os.Chmod(dstFile, srcStat.Mode())
```

Comment:
Do we need to do this in the prepare function? Since we have any pre-defined original permissions, we can `chmod` to desired permissions in the actual tests after the original files are copied to the target directory. e.g. `pushRoot`.

Reply:
Sounds good.